src/Analyzer/FunctionNode.cpp

```diff
@@ -31,6 +32,15 @@ FunctionNode::FunctionNode(String function_name_)
     children[arguments_child_index] = std::make_shared<ListNode>();
 }
 
+const DataTypes & FunctionNode::getExpectedArgumentTypes() const
```

Comment:
After `https://github.com/ClickHouse/ClickHouse/pull/44541` will be merged, better to name this method `getArgumentTypes`.